### PR TITLE
feat: sepolia network

### PR DESCRIPTION
**Pull request by shoom3301**

### CODE REVIEW

File: src/common/consts.ts [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import { SupportedChainId } from './chains'
-import contractNetworks from '@cowprotocol/contracts/networks.json'
-
-const { GPv2Settlement } = JSON.parse(contractNetworks as unknown as string) as typeof contractNetworks
 
 export const BUY_ETH_ADDRESS = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
 export const EXTENSIBLE_FALLBACK_HANDLER = '0x2f55e8b20D0B9FEFA187AA7d00B6Cbe563605bF5'
 export const COMPOSABLE_COW = '0xfdaFc9d1902f4e0b84f65F49f244b32b31013b74'
+const SETTLEMENT_CONTRACT = '0x9008D19f58AAbD9eD0D60971565AA8510560ab41'
```
Comment (shoom3301) [inline src/common/consts.ts:6]: https://github.com/cowprotocol/contracts/blob/main/networks.json

**Comment by shoom3301** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/subgraph/api.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export const SUBGRAPH_PROD_CONFIG: ApiBaseUrls = {
   [SupportedChainId.MAINNET]: SUBGRAPH_BASE_URL + '/cow',
   [SupportedChainId.GNOSIS_CHAIN]: SUBGRAPH_BASE_URL + '/cow-gc',
   [SupportedChainId.GOERLI]: SUBGRAPH_BASE_URL + '/cow-goerli',
+  [SupportedChainId.SEPOLIA]: SUBGRAPH_BASE_URL + '/cow-sepolia',
```
Comment (shoom3301) [inline src/subgraph/api.ts:21]: I haven't tests subgraph endpoints, but I think it's not so critical now

File: src/common/consts.ts [hunk 3]
```diff
@@ -1,50 +1,42 @@
 import { SupportedChainId } from './chains'
-import contractNetworks from '@cowprotocol/contracts/networks.json'
```
Comment (alfetopito) [inline src/common/consts.ts:2]: I assume you are doing this to reduce the lib size?

In one hand, I see it might be an unnecessary bloat. On the other, it's the source of truth.
Ok, right now all the addresses happen to be the same, and we have a limit number of networks

https://github.com/cowprotocol/contracts/blob/main/networks.json#L67-L87

But it's not guaranteed to have always the same address AFAICT, and we might start adding more networks faster in the future.

Still, I see we still depend on the contracts package. Then, what's the benefit of hardcoding the contract address?

File: src/order-book/generated/models/OrderPostError.ts
Comment (alfetopito) [inline src/order-book/generated/models/OrderPostError.ts:1]: The changes here seem unrelated. Where are they coming from?

File: src/subgraph/api.ts (hunk 2 above)
Comment (alfetopito) [reply src/subgraph/api.ts:21]: Can it not be added then?
I rather wait until we have the subgraph deployed to not vendor the sdk with non-working changes.

Comment (shoom3301) [reply src/subgraph/api.ts:21]: I can keep it empty

File: src/common/consts.ts (hunk 3 above)
Comment (shoom3301) [reply src/common/consts.ts:2]: `@cowprotocol/contracts` is not updated yet unfortunately, the latest version doesn't contain config for Sepolia.
It is the main reason why I did this change.

File: src/order-book/generated/models/OrderPostError.ts
Comment (shoom3301) [reply src/order-book/generated/models/OrderPostError.ts:1]: This is a result of `codegen`. So, it's just a generated code, it seems smth changed in the contracts repo

File: src/subgraph/api.ts (hunk 2 above)
Comment (shoom3301) [reply src/subgraph/api.ts]: Done

File: src/order-book/generated/models/OrderPostError.ts
Comment (alfetopito) [reply src/order-book/generated/models/OrderPostError.ts:1]: Ahh ok. Makes sense.

File: src/common/consts.ts (hunk 3 above)
Comment (shoom3301) [reply src/common/consts.ts:2]: >But it's not guaranteed to have always the same address AFAICT, and we might start adding more networks faster in the future.

I promise I will change it here once we have a case :)

Comment (alfetopito) [reply src/common/consts.ts:2]: Hmmm.
In that case, can you make this change in a way that indicates it's only temporary until the contracts package is not up to date?
A TODO, commented out original code, just so it can be brought back in sync later on.

Comment (alfetopito) [reply src/common/consts.ts:2]: Oh, your reply came while I was writing mine :)
image

Sure, I've created an issue to track it https://github.com/cowprotocol/cow-sdk/issues/190

**Review by alfetopito — APPROVED** [review]

**Comment by alfetopito** [comment]
Oh, there's one lint issue 
image
